src/arrayMethodJoin.js

```diff
@@ -4,9 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = String(this[0]);
+
+    if (this[0] === null || this[0] === undefined) {
+      result = '';
+    }
```

Comment:
```suggestion
    let result = '';
```
you have the same logic into the loop
so lets keep our code DRY
remove it here and iterate from the first element